source/extensions/http/header_validators/envoy_default/http1_header_validator.cc

```diff
@@ -287,19 +287,48 @@ Http1HeaderValidator::validateRequestHeaderMap(RequestHeaderMap& header_map) {
       return {RequestHeaderMapValidationResult::Action::Reject,
               UhvResponseCodeDetail::get().InvalidUrl};
     }
-  } else if (!config_.uri_path_normalization_options().skip_path_normalization() &&
-             path_is_absolute) {
-    // TODO(#6589) - Validate and normalize the path, which must be a valid URI. This will be
-    // similar to:
+  } else if (!config_.uri_path_normalization_options().skip_path_normalization()) {
+    // Validate and normalize the path, which must be a valid URI. This is only run if the config
+    // is active.
     //
-    // auto path_result = normalizePathUri(header_map);
-    // if (path_result != RequestHeaderMapValidationResult::Accept) {
-    //   return path_result;
-    // }
+    // If path normalization is disabled then the path will be validated against the RFC character
+    // set in validateRequestHeaderEntry.
+    auto path_result = path_normalizer_.normalizePathUri(header_map);
+    if (!path_result.ok()) {
+      return path_result;
+    }
+
+    path = header_map.path();
   }
 
-  // If path normalization is disabled or the path isn't absolute then the path will be validated
-  // against the RFC character set in validateRequestHeaderEntry.
+  if (!is_connect_method && !path_is_asterisk && path.at(0) != '/') {
+    // The path is most likely in absolute-form, which we need to verify and then use the authority
+    // component as the Host/:authority header. From RFC 9112,
+    // https://www.rfc-editor.org/rfc/rfc9112.html#section-3.2.2:
+    //
+    // When a proxy receives a request with an absolute-form of request-target, the
+    // proxy MUST ignore the received Host header field (if any) and instead replace
+    // it with the host information of the request-target. A proxy that forwards
+    // such a request MUST generate a new Host field-value based on the received
+    // request-target rather than forward the received Host field-value.
+    // ...
+    // If the target URI includes an authority component, then a client MUST send a
+    // field-value for Host that is identical to that authority component,
+    // excluding any userinfo subcomponent and its "@" delimiter (Section 2.7.1).
+    ::Envoy::Http::Utility::Url url;
```

Comment:
In the current architecture this part is in the H/1 [codec](https://github.com/envoyproxy/envoy/blob/9ae348cc43e0341ec7eadd6a4addce587afe1b27/source/common/http/http1/codec_impl.cc#L1067).

I think it makes sense to move it here eventually, but we should do it in a follow up PR, to make it easier to review. So for now we should just codify that UHV expects the path to be in the origin-form unless it is a CONNECT with empty path or OPTIONS with asterisk-form.

In this case we can just check that the pat start with '/' before calling `normalizePathUri` and remove this code, since H/1 code takes care of overwriting the Host header for now.

Reply:
I've removed this.

Reply:
I've removed this.